File: src/main/java/ru/anseranser/peshki/input/MoveCommand.java
package ru.anseranser.peshki.input;

import java.util.List;

public sealed interface MoveCommand {
    record PlacePawn(int playerNumber, int diceValue) implements MoveCommand {}
    record MovePawn(int playerNumber, int pawnNumber, int steps, List<Integer> consumedDice) implements MoveCommand {}
    record PlaceAndMove(int playerNumber, int placeDice, int moveDice) implements MoveCommand {}
}

File: web/src/engine/MoveCommand.ts
export type MoveCommand =
  | { type: 'PlacePawn'; playerNumber: number; diceValue: number }
  | { type: 'MovePawn'; playerNumber: number; pawnNumber: number; steps: number; consumedDice: number[] }
  | { type: 'PlaceAndMove'; playerNumber: number; placeDice: number; moveDice: number };
